src/components/DistanceRequest.js

```diff
@@ -78,7 +86,13 @@ function DistanceRequest({transactionID, transaction, mapboxAccessToken}) {
 
     const waypoints = lodashGet(transaction, 'comment.waypoints', {});
     const numberOfWaypoints = _.size(waypoints);
+
     const lastWaypointIndex = numberOfWaypoints - 1;
+    const isLoadingRoute = lodashGet(transaction, 'comment.isLoading', false);
+    const hasRouteError = Boolean(lodashGet(transaction, 'errorFields.route'));
+    const previousWaypoints = usePrevious(waypoints);
+    const haveWaypointsChanged = !_.isEqual(previousWaypoints, waypoints);
+    const shouldFetchRoute = !haveWaypointsChanged || isOffline || isLoadingRoute || !TransactionUtils.validateWaypoints(waypoints);
```

Comment:
is there a specific reason we're not use `useEffect` with `waypoints` as a dependency?

Reply:
https://github.com/Expensify/App/pull/25434#discussion_r1302152218

Updated it here - mostly so the logic within the useEffect is cleaner

Reply:
ah I meant just doing

```ts
useEffect(() => {
   // fetch route if necessary
   }, [waypoints]
  )
```

Reply:
Hmm I'm not quite following,
GetRoute takes a transactionID so that is required with our linter to add in the useEffect

Do you mean only updating shouldFetchRoute with a useEffect?

Reply:
by the way, please merge this PR when you're ready. this is not a blocker 😄

Reply:
wait, I'm confused on another point. isn't the condition for `shouldFetchRoute` the opposite of the current one?

for example, `shouldFetchRoute` should be `true` when `isOffline` is true?

Reply:
Ah you're right - I messed up the logic when I updated this

Updated the conditions